tests/jvm/src/test/scala/scala/meta/tests/semanticdb/PreciseTypeSuite.scala

```diff
@@ -0,0 +1,78 @@
+package scala.meta.tests.semanticdb
+
+import scala.meta.internal.semanticdb.scalac.SymbolMode
+import scala.meta.internal.semanticdb3.Print
+import scala.meta.internal.{semanticdb3 => s}
+
+class PreciseTypeSuite extends SemanticdbSuite(symbols = SymbolMode.AllWithMaterializedTypes) {
+
+  private def testActualTypes(code: String)(occurencesTypes: String) = test(code){
+    val db = computeDatabaseFromSnippet(code)
+    val allOccurences = db.occurrences ++ db.synthetics.flatMap(_.text).flatMap(_.occurrences)
+
+    def printOccurence(occ: s.SymbolOccurrence) = occ.materializedTpe.map( tpe =>
+          s"${Print.range(occ.range.get)} ${Print.tpe(db, tpe)}"
+      )
```

Comment:
What do you guys think about changing `metap` to always print types associated with occurrences. This will probably be useful for `metap` users and will simplify testing. For example:

Before:
```
Occurrences:
[0:7..0:8): M <= _empty_.M.
```

After:
```
Occurrences:
[0:7..0:8): M <= _empty_.M. (M.type)
```

Reply:
This may easily end up being too verbose, so let's see how this will look in practice.

Reply:
Changes are there and output is quite nice